src/core_plugins/kibana/public/context/components/size_picker/size_picker.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import contextSizePickerTemplate from './size_picker.html';
+import './size_picker.less';
+
+const module = uiModules.get('apps/context', [
+  'kibana',
+]);
+
+module.directive('contextSizePicker', function ContextSizePicker() {
+  return {
+    bindToController: true,
+    controller: ContextSizePickerController,
+    controllerAs: 'contextSizePicker',
+    replace: true,
+    restrict: 'E',
+    scope: {
+      count: '=',
+      setCount: '=',
+    },
+    template: contextSizePickerTemplate,
+  };
+});
+
+
+function ContextSizePickerController() {
+  this.getOrSetCount = (count) => (
+    _.isUndefined(count) ? this.count : this.setCount(count)
```

Comment:
`count` is supposed to be read-only, because I don't want to scatter the state mutation everywhere. The component's concern is twofold:

* display the current value of `count` as passed down from the parent
* report the intention to set a new count to the application

From this it becomes clear, that the directive should leave the decision of what `count` actually is to the application. In this case, the `setCount` action actually clamps the value to [0, inf], which the directive could not know. If was implemented to know this, the business logic would not be DRY and the component tightly coupled to its usage in the application.